armi/physics/neutronics/tests/test_crossSectionManager.py

```diff
@@ -104,21 +105,198 @@ def setUp(self):
+        )
+        self.assertAlmostEqual(newBc.avgNucTemperatures["U235"], 600.0)
+        self.assertAlmostEqual(newBc.avgNucTemperatures["FE56"], expectedIronTemp)
+        self.assertAlmostEqual(newBc.avgNucTemperatures["NA23"], expectedSodiumTemp)
+
+
+class TestComponentAveraging(unittest.TestCase):
+
+    def setUp(self):
+        fpFactory = test_lumpedFissionProduct.getDummyLFPFile()
+        self.blockList = makeBlocks(3)
+        for bi, b in enumerate(self.blockList):
+            b.setType("fuel")
+            b.setLumpedFissionProducts(fpFactory.createLFPsFromFile())
+            # put some trace Fe-56 and Na-23 into the fuel
+            # zero out all fuel nuclides except U-235 (for mass-weighting of component temperature)
+            for nuc in b.getNuclides():
+                b.setNumberDensity(nuc, 0.0)
+            b.setNumberDensity("U235", bi)
+            b.setNumberDensity("FE56", bi / 2.0)
+            b.setNumberDensity("NA23", bi / 3.0)
+            for c in b:
+                if c.hasFlags(Flags.FUEL):
+                    c.temperatureInC = 600.0 + bi
+                elif c.hasFlags([Flags.CLAD, Flags.DUCT, Flags.WIRE]):
+                    c.temperatureInC = 500.0 + bi
+                elif c.hasFlags([Flags.BOND, Flags.COOLANT, Flags.INTERCOOLANT]):
+                    c.temperatureInC = 400.0 + bi
+
+        self.bc = AverageBlockCollection(
+            self.blockList[0].r.blueprints.allNuclidesInProblem
+        )
+        self.bc.extend(self.blockList)
+
+    def test_getAverageComponentNumberDensities(self):
+        """
+        Test component number density averaging
+        """
+        # becaue of the way densities are set up, the middle block (index 1 of 0-2) component densities are equivalent to the average
+        b = self.bc[1]
+        for compIndex, c in enumerate(b.getComponents()):
+            avgDensities = self.bc._getAverageComponentNumberDensities(compIndex)
+            compDensities = c.getNumberDensities()
+            for nuc in c.getNuclides():
+                self.assertAlmostEqual(
+                    compDensities[nuc],
+                    avgDensities[nuc],
+                    msg=f"{nuc} density {compDensities[nuc]} not equal to {avgDensities[nuc]}!",
+                )
+
+    def test_getAverageComponentTemperature(self):
+        """
+        Test mass-weighted component temperature averaging
+        """
+        b = self.bc[0]
+        massWeightedIncrease = 5.0 / 3.0
+        baseTemps = [600, 400, 500, 500, 400, 500, 400]
+        expectedTemps = [t + massWeightedIncrease for t in baseTemps]
+        for compIndex, c in enumerate(b.getComponents()):
+            avgTemp = self.bc._getAverageComponentTemperature(compIndex)
+            self.assertAlmostEqual(
+                expectedTemps[compIndex],
+                avgTemp,
+                msg=f"{c} avg temperature {avgTemp} not equal to expected {expectedTemps[compIndex]}!",
+            )
```

Comment:
The test you provided here doesn't give coverage for the first block of the conditional (`if avgComponentMass = 0.0`) in getAverageComponentTemperature. If you swap out `b` with `test_block.loadTestBlock` and self.bc` with a local instance of `AverageBlockCollection`, then you can get full coverage.

Reply:
I'm pretty sure it does cover it. I tested it by putting a print statement in the `if avgComponentMass == 0.0` block and it does enter the block.

Reply:
There is literally another test to cover this exact thing via test_getAverageComponentTemperatureNoMass`. 

Doh!